insonmnia/miner/salesman.go

```diff
@@ -0,0 +1,345 @@
+		if plan.DealID.String() == dealID {
+			return plan, nil
+		}
+	}
+	return nil, fmt.Errorf("ask plan for deal id %s is not found", dealID)
+}
+
+func (m *Salesman) syncRoutine(ctx context.Context) {
+	m.log.Debugf("starting sync routine")
+	ticker := util.NewImmediateTicker(time.Second)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ticker.C:
+			m.syncWithBlockchain(ctx)
+		case <-ctx.Done():
+			return
+		}
+	}
+}
+
+func (m *Salesman) syncWithBlockchain(ctx context.Context) {
+	m.log.Debugf("syncing salesman with blockchain")
+	plans := m.state.AskPlans()
+	for _, plan := range plans {
+		orderId := plan.GetOrderID()
+		dealId := plan.GetDealID()
+		if !dealId.IsZero() {
+			m.checkDeal(ctx, plan)
+		} else if !orderId.IsZero() {
+			m.checkOrder(ctx, plan)
+		} else if dealId.IsZero() && orderId.IsZero() {
+			m.placeOrder(ctx, plan)
+		}
+	}
+}
+
+func (m *Salesman) restoreState() error {
+	askPlans := m.state.AskPlans()
+	for _, plan := range askPlans {
+		m.resources.Consume(plan)
+	}
+	//TODO: restore tasks
+	return nil
+}
+
+func (m *Salesman) checkDeal(ctx context.Context, plan *sonm.AskPlan) {
+	m.log.Debugf("checking deal %s for ask plan %s and order %s",
+		plan.DealID.Unwrap().String(), plan.ID, plan.GetOrderID().Unwrap().String())
+	deal, err := m.eth.Market().GetDealInfo(ctx, plan.GetDealID().Unwrap())
+	if err != nil {
+		m.log.Warnf("could not get deal info for order %s, ask %s - %s",
+			plan.GetOrderID().Unwrap().String(), plan.ID, err)
+		// TODO: log, what else can we do?
+		return
+	}
+
+	if deal.Status == sonm.DealStatus_DEAL_CLOSED {
+		err = m.resources.Release(plan.ID)
+		if err != nil {
+			m.log.Warnf("could not release resources for deal %s, order %s, ask %s - %s",
+				plan.DealID.Unwrap().String(), plan.OrderID.Unwrap().String(), plan.ID, err)
+		}
+		plan.DealID = nil
+		plan.OrderID = nil
+		m.state.SaveAskPlan(plan)
+	}
+}
+
+func (m *Salesman) checkOrder(ctx context.Context, plan *sonm.AskPlan) {
+	//TODO: validate deal that it is ours
+	m.log.Infof("checking order %s for ask plan %s", plan.GetOrderID().Unwrap().String(), plan.ID)
+	order, err := m.eth.Market().GetOrderInfo(ctx, plan.GetOrderID().Unwrap())
+	if err != nil {
+		m.log.Warnf("could not get order info for order %s - %s", plan.GetOrderID().Unwrap().String(), err)
+		// TODO: log, what else can we do?
+		return
+	}
+
+	if !order.DealID.IsZero() {
+		deal, err := m.eth.Market().GetDealInfo(ctx, order.DealID.Unwrap())
+		if err != nil {
+			m.log.Warnf("could not get deal info for ID %s from market - %s", order.DealID.Unwrap().String(), err)
+			// TODO: log, what else can we do?
+			return
+		}
+		m.dealChan <- deal
+
+		plan.DealID = order.DealID
+
+		if err := m.state.SaveAskPlan(plan); err != nil {
+			m.log.Warnf("could not get save ask plan with deal %s - %s", order.DealID.Unwrap().String(), err)
+			// TODO: log, what else can we do?
+			return
+		}
+	}
+}
+
+func (m *Salesman) placeOrder(ctx context.Context, plan *sonm.AskPlan) {
```

Comment:
Why can't we return error here?

Reply:
it's a private function and we can not do anything with this error, but ok, let's make it more idiomatic